packages/block-editor/src/components/rich-text/index.js

```diff
@@ -229,6 +235,7 @@ export function RichTextWrapper(
 				selection.start = {
 					clientId,
 					attributeKey: identifier,
+					[ instanceIdKey ]: instanceId,
```

Comment:
If there is a valid `identifier`, the `instanceId` is redundant. It's better to always set only one of them, not both.

Reply:
```js
const [ idKey, idValue ] = identifier
  ? [ 'attributeKey', identifier ]
  : [ instanceIdKey, instanceId ];

selection.start = {
  [ idKey ]: idValue
}
```